### PR TITLE
Load CT and dose data

### PR DESCRIPTION
**Pull request by YingClara92**

### CODE REVIEW

File: DataGenerator/DataGenerator.py [hunk 1]
```diff
@@ -18,180 +18,222 @@
...
-        self.transform        = transform
-        self.target_transform = target_transform
-        self.label            = label
-        self.inference        = inference
-        self.PatientList      = PatientList
-        self.n_norm = n_norm
-        self.c_norm = c_norm
-        self.config = config
+        self.transform = transform
+        self.config = kwargs['config']
```
Comment (cacof1) [inline DataGenerator/DataGenerator.py:30]: You shouldn't use the kwargs argument here. You should define config in the function declaration

File: DataGenerator/DataGenerator.py [hunk 2]
```diff
@@ -18,180 +18,222 @@
...
         datadict = {}
         roiSize = [10, 40, 40]
-        AnatomyPath = PatientList[i].scan.experiment['CT'].path()
-        
-        #label    = self.mastersheet[self.label].iloc[id]
+        patient_id = self.PatientList[id].label
+        ScanPath = self.config['DATA']['DataFolder'] + patient_id + '\\' + patient_id + '\\' + 'scans\\'
+        subfolder_list = os.listdir(ScanPath)
+        reader = image_reader.ITKReader()
+        label = self.PatientList[id].fields['survival_months']
```
Comment (cacof1) [inline DataGenerator/DataGenerator.py:50]: Here the label should point to what is in the config

File: DataGenerator/DataGenerator.py [hunk 3]
```diff
@@ -18,180 +18,222 @@
...
-            #datadict["Dose"]  = np.expand_dims(CropImg(dose, maxDoseCoords, roiSize),0)
+            Dose_match_folders = [match for match in subfolder_list if "Dose" in match]
+            full_Dose_path = [ScanPath + Dose_match_folder + '\\resources\\DICOM\\files\\' for Dose_match_folder in
+                              Dose_match_folders]
+            if len(full_Dose_path) > 1:
+                for dose_path in full_Dose_path:
+                    itkObj = reader.read(dose_path + '1-1.dcm')
+                    dose, info = reader.get_data(itkObj)
+                    for value in info.values():
+                        if 'totalhetero' in value:
```
Comment (cacof1) [inline DataGenerator/DataGenerator.py:71]: This is one of my point, we shouldn't use hard coded value like 'totalhetero'. Perhaps a better way to describe the toal dose.

File: DataGenerator/DataGenerator.py [hunk 4]
```diff
@@ -18,180 +18,222 @@
...
-    def train_dataloader(self): return DataLoader(self.train_data, batch_size=self.batch_size, shuffle=True, num_workers=0, collate_fn=custom_collate)
-    def val_dataloader(self):   return DataLoader(self.val_data,   batch_size=self.batch_size, num_workers=0, collate_fn=custom_collate)
-    def test_dataloader(self):  return DataLoader(self.test_data,  batch_size=self.batch_size, collate_fn=custom_collate)
 
 def QueryFromServer(config, **kwargs):
     print("Querying from Server")
     ## Get List of Patients
-    session  = xnat.connect('http://128.16.11.124:8080/xnat/', user=config["SERVER"]["User"], password='yzhan')
-    project  = session.projects[config["SERVER"]["Project"]]    
+    session = xnat.connect('http://128.16.11.124:8080/xnat/', user=config["SERVER"]["User"], password='yzhan')
```
Comment (cacof1) [inline DataGenerator/DataGenerator.py:154]: please change your pword to that of the config

File: Training/Regression.py [hunk 5]
```diff
@@ -65,7 +65,7 @@
 label = config['DATA']['target']
 
 PatientList = QueryFromServer(config)
-SynchronizeData(config, PatientList)
+# SynchronizeData(config, PatientList)
```
Comment (cacof1) [inline Training/Regression.py:68]: That's probably useful

File: DataGenerator/DataGenerator.py [hunk 6]
```diff
@@ -21,20 +22,20 @@
...
 class DataGenerator(torch.utils.data.Dataset):
-    def __init__(self, PatientList, transform=None, **kwargs):
+    def __init__(self, PatientList, config, keys, transform=None, **kwargs):
         super().__init__()
         self.transform = transform
-        self.config = kwargs['config']
-        self.keys = kwargs['keys']
-        # self.label            = label
+        self.keys = keys
         self.inference = kwargs['inference']
```
Comment (cacof1) [inline DataGenerator/DataGenerator.py:34]: Idem as said last time

File: DataGenerator/DataGenerator.py [hunk 7]
```diff
@@ -18,180 +19,253 @@
...
+        correct_Origin = reader.read(full_CT_path + dicom_files[0])
+        itkObj = reader.read(full_CT_path)
+        itkObj.SetOrigin(correct_Origin.GetOrigin())
         # Get the mask of PTV
         if "Dose" in self.keys or "Anatomy" in self.keys:
             if self.config['DATA']['Use_mask']:
-                path = self.mastersheet["CTPath"].iloc[id]
-                mask = path[0:-16] + 'structs\PTV.nrrd'
-                mask_img = LoadImg(mask)
+                RT_match_folder = [match for match in subfolder_list if "Structs" in match]
```
Comment (cacof1) [inline DataGenerator/DataGenerator.py:61]: Both for CT and structs, there shouldn't be a "matching parameter". We have to work with Olivia to have consistency between the naming of the different modalities and call them directly.

File: DataGenerator/DataGenerator.py [hunk 8]
```diff
@@ -18,180 +19,253 @@
...
+        label = self.PatientList[id].fields[self.config['DATA']['target']]
+        CT_match_folder = [match for match in subfolder_list if "CT" in match]
+        full_CT_path = ScanPath + CT_match_folder[0] + '\\resources\\DICOM\\files\\'
+        dicom_files = os.listdir(full_CT_path)
+        correct_Origin = reader.read(full_CT_path + dicom_files[0])
+        itkObj = reader.read(full_CT_path)
+        itkObj.SetOrigin(correct_Origin.GetOrigin())
         # Get the mask of PTV
         if "Dose" in self.keys or "Anatomy" in self.keys:
             if self.config['DATA']['Use_mask']:
```
Comment (cacof1) [inline DataGenerator/DataGenerator.py:60]: Please change to a single tag

**Review by cacof1 — COMMENTED** [review]
Please look into my specific comments

**Review by cacof1 — APPROVED** [review]
